src/fsharp/CompileOps.fs

```diff
@@ -2000,11 +2000,12 @@ let ResolveFileUsingPaths(paths, m, name) =
         let searchMessage = String.concat "\n " paths
         raise (FileNameNotResolved(name, searchMessage, m))            
 
-let GetWarningNumber(m, s:string) =
+let GetWarningNumber(s:string) =
     try 
-        Some (int32 s)
-    with err -> 
-        warning(Error(FSComp.SR.buildInvalidWarningNumber(s), m))
+        // Trim off leading "FS" to allow ""/warnon:FS0001;FS0002;0003; anything else we ignore
+        let number = if s.StartsWith("FS", StringComparison.InvariantCulture) = true then s.Substring(2) else s
```

Comment:
so this will detect ``FSxxx``, but ignore ``fsxxx``, similar to C#?
I'm all for C# compatibility, but we don't need to duplicate their bugs...

Reply:
To be completely honest: 
- It is fine to accept "FS" or even "fs" prefix
- But the compiler should still warn when it gets invalid values like NU1234 those need to be filtered in the SDK
- What they have done with the C# compiler is wrong and feels more like a quick hack...

Reply:
But @matthid don't forget it's so "convenient" to just hijack things.

Reply:
@matthid the C# compiler thing may be a bug but it's not a hack.  I expect it does what it does because the original native compiler did it this way.  That's usually why it does unexpected things.

I have some sympathy with the splitting NU0000 in the targets file rather than the compiler, although there is a lot of benefit in just using the same approach as C#, it means if you learn something for one language it works the same in another.

@forki sometimes it is convenient to extend a generalization.  The <NoWarn> property in a build file doesn't specify only do it for F# compiler invocations.  It's not unreasonable to make more of the build tools aware of it.  If the F# compiler had behaved similarly to C# originally, no one would have mentioned it.